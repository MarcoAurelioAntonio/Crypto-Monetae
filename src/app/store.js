import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from '../features/Card/cardsSlice';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});

export default store;
