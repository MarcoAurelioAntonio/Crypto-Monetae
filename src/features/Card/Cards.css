@import url('https://fonts.googleapis.com/css2?family=Viaoda+Libre&display=swap');

.main-section {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  margin: 0;
  width: 100vw;
  min-height: 100vh;
  height: auto;
  background-image: url(../assets/mobile-bacground.jpg);
  background-size: cover;
  background-repeat: repeat-y;
  padding-top: 100px;
}

.search-box {
  width: 40vw;
  padding: 10px;
  padding-left: 30px;
  font-size: 20px;
  border: none;
  margin-bottom: 30px;
  color: rgb(2, 2, 193);
  font-weight: 700;
}

.input-section {
  width: 100vw;
  height: 70px;
  margin-top: 25px;
}

::placeholder {
  color: rgb(100, 100, 255);
  font-style: italic;
  font-weight: 300;
  opacity: 0.8;
}

.search-box:focus {
  outline: none;
  background-color: aliceblue;
}

@media (max-width: 600px) {
  .search-box {
    width: 70vw;
  }
}
