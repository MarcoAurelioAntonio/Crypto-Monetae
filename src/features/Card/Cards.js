import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCoinsAsync } from './cardsSlice';
import Card from './Card';
import './Cards.css';

export default function Cards() {
  const dispatch = useDispatch();

  const [searchcoin, setSearchcoin] = useState('');
  const onSearchItem = (e) => {
    setSearchcoin(e.target.value);
  };

  const coinsData = useSelector((state) => state.coins);

  const searchItem = coinsData.filter(
    (filteredItem) => filteredItem.name.toLowerCase().includes(searchcoin.toLowerCase())
    || filteredItem.symbol.toLowerCase().includes(searchcoin.toLowerCase()),
  );

  useEffect(() => {
    if (searchItem.length === 0) {
      dispatch(fetchCoinsAsync());
    }
  }, []);

  console.log(coinsData);

  return (
    <div className="main-section">
      <div className="input-section">
        <input
          className="search-box"
          placeholder="Enter Crypto name"
          type="text"
          value={searchcoin}
          onChange={onSearchItem}
        />
      </div>
      {searchItem.map((coin) => (
        <Card key={coin.id} coins={coin} />
      ))}
    </div>
  );
}
