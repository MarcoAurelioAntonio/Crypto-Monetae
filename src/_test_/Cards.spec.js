import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Cards from '../features/Card/Cards';

it('Cards page rendering', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Cards />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
