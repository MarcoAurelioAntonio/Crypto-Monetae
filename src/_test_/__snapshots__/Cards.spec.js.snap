// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards page rendering 1`] = `
<div
  className="main-section"
>
  <div
    className="input-section"
  >
    <input
      className="search-box"
      onChange={[Function]}
      placeholder="Enter Crypto name"
      type="text"
      value=""
    />
  </div>
</div>
`;
